packages/protocol/contracts/rrp/authorizers/Api3TokenLock.sol

```diff
@@ -0,0 +1,393 @@
+            ERROR_INSUFFICIENT_AMOUNT
+        );
+        require(
+            !clientAddressToBlacklistStatus[_clientAddress],
+            ERROR_CLIENT_BLACKLISTED
+        );
+        uint256 withdrawAmount = tokenLocks[_airnodeId][_clientAddress]
+        .lockAmountAt[msg.sender] - lockAmount;
+        tokenLocks[_airnodeId][_clientAddress].lockAmountAt[
+            msg.sender
+        ] = lockAmount;
+        assert(IApi3Token(api3Token).transfer(msg.sender, withdrawAmount));
+
+        emit Withdraw(_airnodeId, _clientAddress, msg.sender, withdrawAmount);
+    }
+
+    /// @notice Transfers the user's lock period and tokens to another client
+    /// @param _airnodeId The airnode id
+    /// @param _fromClientAddress The client from which the transfer will be done
+    /// @param _toClientAddress The target client to which the transfer will be done
+    function transfer(
+        bytes32 _airnodeId,
+        address _fromClientAddress,
+        address _toClientAddress
+    ) external override {
+        AirnodeClient storage from = tokenLocks[_airnodeId][_fromClientAddress];
+        AirnodeClient storage to = tokenLocks[_airnodeId][_toClientAddress];
+        require(
+            from.lockAmountAt[msg.sender] != 0,
+            "locked amount must be != 0"
+        );
+        require(to.lockAmountAt[msg.sender] == 0, "locked amount must be 0");
+        require(
+            !clientAddressToBlacklistStatus[_fromClientAddress],
+            "From Client blacklisted"
+        );
+        require(
+            !clientAddressToBlacklistStatus[_toClientAddress],
+            "To Client blacklisted"
+        );
+
+        uint256 amount = from.lockAmountAt[msg.sender];
+        from.lockAmountAt[msg.sender] = 0;
+        to.lockAmountAt[msg.sender] = amount;
+
+        uint256 expirationTime = from.canUnlockAt[msg.sender];
+        from.canUnlockAt[msg.sender] = 0;
+        to.canUnlockAt[msg.sender] = expirationTime;
+
+        from.whitelistCount--;
+        if (from.whitelistCount == 0) {
+            IApi3RequesterRrpAuthorizer(api3RequesterRrpAuthorizer)
+                .setWhitelistExpiration(_airnodeId, _fromClientAddress, 0);
+        }
+
+        to.whitelistCount++;
+        if (to.whitelistCount == 1) {
+            IApi3RequesterRrpAuthorizer(api3RequesterRrpAuthorizer)
+                .setWhitelistExpiration(
+                _airnodeId,
+                _toClientAddress,
+                type(uint64).max
+            );
+        }
+
+        emit Transfer(
+            _airnodeId,
+            _fromClientAddress,
+            _toClientAddress,
+            msg.sender,
+            amount,
+            expirationTime
+        );
+    }
+
+    /// @notice Burns tokens for a user of an airnode-client pair.
+    /// Can only be done when the client is blacklisted.
+    /// @param _airnodeId The airnode id
+    /// @param _clientAddress The client address
+    /// @param _burnTarget The address of the user
+    function burn(
```

Comment:
Does it make sense that anyone can call this function? I'm thinking that maybe only an Admin should be allowed to call it since it looks like a cleanup task to remove token locks for blacklisted requesters.